### ISSUE TITLE
[platform/upgrade] Available source/destination version upgrades inconsistent

### ISSUE THREAD

**Issue by joeybenamy** [airbytehq/airbyte#35207]
### Helm Chart Version

0.50.20

### What step the error happened?

Other

### Revelant information

We are running 3 Airbyte OSS clusters in Kubernetes with Helm.  Each instance is in a separate Kubernetes cluster and they don't share any resources.  All 3 of our environments run the same configuration and AIrbyte versions, but each instance shows different versions of various source and destination packages as being the latest versions.  In some environments, the `Upgrade all` button for Sources and Destinations also does not always upgrade the packages to the same versions in each environment because it thinks the latest version is a different version.  It seems that this is a function of when the helm chart is installed.  If we nuke all 3 environments and reinstall them at the same time, the latest versions of the packages match.  We expect this to be consistent between environments regardless of when it was deployed.  Some examples:
MySQL source:
* UAT - latest version is 3.3.4
* STAGING - latest version is 3.3.5
* PROD - latest version is 3.3.5

Snowflake destination:
* UAT - latest version is 3.5.11
* STAGING - latest version is 3.5.9
* PROD - latest version is 3.5.10

**Comment by pmossman** [comment]
Refining notes:
- Might be worth discussing w/ @pedroslopez when he's back, keeping in inbox for now

**Comment by pmossman** [comment]
Refining notes:
- Building Airbyte determines your initial catalog. So build once and using those images across several environments SHOULD provide the same initial versions across instances
- For _latest_ versions, the Settings page will make an API request to our registry on page load. 
  - There may be some cacheing on this call, so _potentially_ if we aren't clearing that cache, we could serve different results for different instances 
- TLDR: We think our system should be working as requested by this user, so there may be a bug that is worth investigating

**Comment by octavia-squidington-iii** [comment]
This issue was closed because it has been inactive for 20 days since being marked as stale.
